polar-core/src/vm.rs

```diff
@@ -1406,9 +1406,34 @@ impl PolarVirtualMachine {
             Value::Expression(_) => {
                 return self.query_for_operation(&term);
             }
+            Value::Variable(_a_symbol) => {
+                let val = self.deref(term);
+
+                if val == *term {
+                    // variable was unbound
+                    // apply a constraint to variable that it must be truthy
+                    self.push_goal(Goal::Query {
```

Comment:
You could do this more directly by pushing a `Goal::Unify` instead.

Reply:
resolved by latest commit